volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -26,43 +26,113 @@
 
 
 class module(generic.GenericIntelProcess):
+    @property
+    def mod_mem_type(self):
+        """Return the mod_mem_type enum choices if available or None if not"""
+        if not self.has_member("_mod_mem_type"):
+            try:
+                vmlinux = linux.LinuxUtilities.get_module_from_volobj_type(
+                    self._context, self
+                )
+                # mod_mem_type and module_memory were added in kernel 6.4 which replaces
+                # module_layout for storing the information around core_layout etc.
+                # see commit ac3b43283923440900b4f36ca5f9f0b1ca43b70e for more information
+                self._mod_mem_type = vmlinux.get_enumeration("mod_mem_type").choices
+            except exceptions.SymbolError:
+                vollog.debug(
+                    f"Unable to find mod_mem_type enum. This is expected on kernels <6.4 but may cause issues with later kernels"
+                )
+                self._mod_mem_type = None
+        return self._mod_mem_type
+
     def get_module_base(self):
-        if self.has_member("core_layout"):
-            return self.core_layout.base
+        if self.mod_mem_type:
```

Comment:
Something about this just seems ugly, but I'm not sure of a better way to do it?  I was thinking of something involving default lookups from `mod_mem_type` or from `mem`, but I can't figure a good way of getting the right results returned more efficiently, so this is probably the right way of doing it (even if something about it doesn't quite sit right to me).  5;P